dashboard/test/controllers/aichat_controller_test.rb

```diff
@@ -232,4 +241,28 @@ class AichatControllerTest < ActionController::TestCase
       as: :json
     assert_equal session.id, json_response['session_id']
   end
+
+  test 'Bad request if required params are not included for student_chat_history' do
+    sign_in(@genai_pilot_teacher)
+    post :chat_completion, params: {studentId: @genai_pilot_student.id}, as: :json
+    assert_response :bad_request
+  end
+
+  test 'pilot student does not have access to student_chat_history test' do
+    sign_in(@genai_pilot_student)
+    post :student_chat_history, params: @valid_params_student_chat_history, as: :json
+    assert_response :forbidden
+  end
+
+  test 'pilot teacher has access to student_chat_history if teacher of student' do
+    sign_in(@genai_pilot_teacher)
+    post :student_chat_history, params: @valid_params_student_chat_history, as: :json
+    assert_response :success
```

Comment:
Any way we can create some mock chat events and verify that we get the right list back?

Reply:
Sure! I added some tests for `log_chat_event` in https://github.com/code-dot-org/code-dot-org/pull/60132 which is not yet merged, so if you're okay with this I'll  add a a test in the follow-up to this PR.

Reply:
@sanchitmalhotra126 FYI, added unit test to check list in https://github.com/code-dot-org/code-dot-org/pull/60322